docs/reference-capabilities/default-capabilities.md

```diff
@@ -0,0 +1,233 @@
+
+_Tip_: Instead of using the default `ref` type for constructors, consider using `iso`. Recall that constructors, like `consume` return their values as an ephemeral type. That means that a constructor typed as `new iso` actually returns a `iso^`. And `iso^` has the terrific property of being convertible to any other type! Consider:
+
+```pony
+class MyIsoClass
+    let foo: String
+
+    new iso create(foo': String) =>
+        foo = foo'
+```
+
+This class can be constructed as a ref or val or whatever else:
+
+```pony
+a_ref: MyIsoClass ref = MyIsoClass.create()
+a_val: MyIsoClass val = MyIsoClass.create()
+```
+
+In fact, this is what happens if you do not specify a constructor at all. The hidden constructor for this class also returns an `iso^`:
+
+```pony
+class AllDefaults
+    var foo: String = "hello"
+```
+
+You can construct instances of this class with a specific capability by specifying the type of the result:
+
+```pony
+let a = AllDefaults // a is iso
+let b: AllDefaults ref = AllDefaults // b is ref
+let c: AllDefaults val = AllDefaults // c is val
+```
+
+Constructors create objects, but what about when you access functions on those objects?
+
+## Default receiver capabilities
+
+Recall that when you call a function or behaviour, it always implicitly gets a reference to the `this` object.
+
+The default receiver capability for a function is `box`, as shown in `get_foo` below:
+
+```pony
+class MyClass
+    var foo: String = ""
+
+    fun get_foo() => foo
+
+    fun ref set_foo(foo': String) => foo = foo'
+```
+
+Notably, you can't mutate a `box`, so mutable methods must be defined as `fun ref` or `fun trn`.
+
+The _only_ behavior capability for an actor is `ref`. You cannot specify a `be val` for example, or even `be ref`. This may be counter-intuitive since the _caller_ only has a tag to the actor, but the actor is totally free to reference its own fields from the behaviour.
+
+## Default recover capabilities
+
+If you are recovering a value, it will default to `val` if the value is immutable (`val` or `box`), and `iso` if it is mutable (`ref` or `trn`):
+
+```pony
+var i_am_iso = recover MyClass.create_ref("abc") end
+var i_am_val = recover MyClass.create_box("def") end
+```
+
+You can, of course, change the capability using `recover ref` or similar.
+
+## Default object literal capabilities
+
+The default capabilities on an object literal depend on whether the object literal specifies a primitive, actor or class, and use the same defaults as the accompanying type:
+
+```pony
+let myObject = object
+    let s: String = "hello"
+end
+```
+
+has `ref` capabilities (it is an object of an anonymous class).
+
+```pony
+let myActor = object
+    be apply() => None
+end
+```
+
+has tag capabilities.
+
+```pony
+let myPrimitive = object
+fun apply() => None
+end
+```
+
+has `val` capabilities.
```

Comment:
```suggestion
has `val` capabilities as it is an anonymous primitive due to having functions, no behaviors, and no internal state.
```

Reply:
This isn't correct. There's no such thing as an anonymous primitive. Is this a primitive?

```pony
class Foo
  fun apply(): U64 =>
    3
```

No, its a class. But it has functions, no behaviors, and no internal state.